SchoolProject/src/main/java/Main.java

```diff
@@ -1,51 +1,113 @@
+
+        var view = new View();
+
+        //Create repositories
+        var studentRepository = new StudentRepositoryImpl();
+        var teacherRepository = new TeacherRepositoryImpl();
+        var subjectRepository = new SubjectRepositoryImpl();
+
+
+        //Read data from json files
+        var readDataService = new ReadDataService();
+        var students = readDataService.readDataFromJson(studentJsonPath, Student[].class);
+        var teachers = readDataService.readDataFromJson(teacherJsonPath, Teacher[].class);
 
-        Scanner scanner = new Scanner(System.in);
-        var school = new School("American School");
 
-        System.out.println("=============================");
-        System.out.println("Welcome to " + school.getName());
-        System.out.println("=============================");
+        //Add data to repositories
+        WriteDataService writeDataService = new WriteDataService();
+        writeDataService.writeData(students,studentRepository);
+        writeDataService.writeData(teachers,teacherRepository);
+        for (Teacher teacher : teachers) {
+            teacher.getSubjectList().forEach(subjectRepository::save);
+        }
+        //See data in repositories
+        //view.showMessage(studentRepository.findAll().toString());
+        //view.showMessage(teacherRepository.findAll().toString());
+        //view.showMessage(subjectRepository.findAll().toString());
+
+        //Create school
+        var school = new School("American School");
 
-        System.out.println("Select your role please: ");
-        System.out.println("1. Teacher");
-        System.out.println("2. Student");
+        view.showMenu(school.getName());
+        String option = view.getString();
 
-        String option = scanner.nextLine();
         boolean access = false;
 
         while (!access) {
-            System.out.println("Please enter your code");
-            AccessValidator accessValidator = new AccessValidator(studentRepository);
+            view.showMessage("Please enter your code");
+
 
-            String code = scanner.nextLine();
+            String code = view.getString();
 
             switch (option) {
                 case "1":
+                    var accessValidator = new AccessValidator(teacherRepository);
                     var teacher = accessValidator.verifyTeacherAccess(code);
                     if (teacher != null) {
                         access = true;
+                        view.showMessage("Welcome " + teacher.getName());
+                        //List the students of "ELEMENTARY_SCHOOL" grade
+                        //Enter a grade for a student
+                        view.showMessage("1. List the students of a grade");
+                        view.showMessage("2. Enter a grade for a student");
+
+                        String option2 = view.getString();
+
+                        switch (option2) {
+                            case "1":
```

Comment:
Maybe you can apply the command pattern here, for the different options in a menu